### ISSUE TITLE
theme.json styles.elements.button.:hover styles generate selector with unscoped `.wp-element-button` selector

### ISSUE THREAD

**Issue by ethanclevenger91** [WordPress/gutenberg#48901]
### Description

Styles defined in `theme.json` for elements > button > :hover result in a selector that includes a non-hovered version of `.wp-element-button`.

If the theme provides non-hover styles for blocks > core/button, those will override the non-block selector and you won't notice. But if the theme exclusively makes use of the elements selector, you will notice on the front-end.

This issue seems to have been introduced in 15.2.0 and persists in 15.2.4.

### Step-by-step reproduction instructions

1. Install twentytwentytwo 1.3 and Gutenberg 15.2.0
2. Create a page with a button
3. View page. Note appearance.
4. In `theme.json`, add the following to `styles.elements`:
```
"button": {
	":hover": {
		"color": {
			"background": "blue"
		}
	}
}
```
5. View page. At first, nothing seems wrong. Hover state works as expected.
6. Inspect button element. Note the following styles is now applied, which includes an unexpected unscoped `.wp-element-button` selector:
```
.wp-element-button, .wp-block-button__link:hover {
  background-color: blue;
}
```
7. Completely remove `styles.blocks.core/button` from `theme.json`
8. View page.
9. Note button is now permanently blue, rather than the default gray.

### Environment info

WordPress 6.1.1, twentytwentytwo 1.3, Gutenberg 15.2.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
Thanks for the report and for the testing steps, @ethanclevenger91 

So far, I haven't been able to replicate the issue. At this step:

> Note button is now permanently blue, rather than the default gray.

On my end the button did **not** stay blue, either in the default or hover state.

### Tested with:

WordPress 6.1.1
Gutenberg 15.2.0 – both active and not active
Gutenberg 15.2.4 – both active and not active
No other active plugins
Twenty Twenty-Two 1.3

**Comment by t-hamano** [comment]
Hi @ethanclevenger91,

I reported a similar problem in #48759. This issue has been fixed in #48759. Since this PR is included in Gutenberg version 15.4, you will be able to confirm that the problem will not occur once that version is released.
